### PR TITLE
TRAFODION-1636 & TRAFODION-1642

### PR DESCRIPTION
**Pull request by robertamarton**
TRAFODION-1636: wms and dcs directories contain NOTICE and LICENSE FILES
TRAFODION-1642: Release 1.3.0 copyrights and license files are incorrect

Fixed up copyrights

Removed Apache copyrights (should only have one copyright in file)
  core/sql/common/swscanf.cpp
  core/sql/common/sqsprintf.cpp
  core/sql/parser/ulexer.h

Removed copyright - deemed to be our code and can safely remove
  core/conn/jdbc_type4/src/org/trafodion/jdbc/t4/T4DatabaseMetaData.java
  core/sql/regress/newregr/card/runregr_card.ksh
  dcs/src/main/java/org/trafodion/dcs/util/ByteBufferUtils.java
  dcs/src/main/java/org/trafodion/dcs/util/SqlUtils.java

Replaced Hewlett Packard copyright with Apache copyright
  core/sql/regress/seabase/EXPECTED021
  core/sql/regress/udr/EXPECTED100.SB
  docs/client_install/source/Trafodion_Client_Installation_Guide.xml
  docs/client_install/source/xml-profile.xml
  docs/command_interface/source/Trafodion_Command_Interface_Guide.xml
  docs/command_interface/source/xml-profile.xml
  docs/odb_manual/Trafodion_odb_User_Guide.xml
  docs/sql_reference/source/Trafodion_SQL_Reference_Manual.xml
  docs/sql_reference/source/xml-profile.xml
  wms/build.xml

Removed code/files that are not being used that contain invalid copyrights
  core/sqf/export/lib/Vili/Functions.pm
  core/sqf/export/lib/Vili/Helper.pm
  core/sqf/export/lib/Vili/NoDriftTimer.pm
  core/sql/common/from_GB2312.c
  core/sql/regress/newregr/mvs/BIG/expectedmv215_RU
  core/sql/regress/newregr/mvs/EXPECTEDMV500
  core/sql/regress/newregr/mvs/EXPECTEDMV500A
  core/sql/regress/newregr/mvs/EXPECTEDMV500A_LINUX
  core/sql/regress/newregr/mvs/TPCD/DBGEN_HELP.TXT
  dcs/src/main/resources/dcs-webapps/master/css/stylesheet.css

Removed filters for Hewlett Packard, Tandem, Compaq, and Microsoft.  None of
test software should returns these values
  core/sql/regress/tools/regress-filter
  core/sql/regress-tools/regress-filter-linux

Removed LICENSE and NOTICE files from child directories
  dcs/LICENSE.txt
  dcs/NOTICE.txt
  wms/LICENSE.txt
  wms/NOTICE.txt
  install/README.rst

Reworded comment so Copright would not show up on search
  core/sql/update/CopyrightCheck.py

### CODE REVIEW

File: core/sql/parser/ulexer.h [hunk 1]
```diff
@@ -47,6 +17,13 @@
 // this software without specific prior written permission.
 
 
+// ULexer.h -- define interfaces for Unicode lexical analyzer class (tcr)
+
+// Flex (version 2.5.4a and earlier) does not support Unicode. Our attempts
+// at extending flex to generate a Unicode scanner were unsuccessful. So, we 
+// hand-code a Unicode scanner for SQL/MX but retain the flex C++ scanner
```
Comment (DaveBirdsall) [inline core/sql/parser/ulexer.h]: Should we say, "Trafodion" instead of "SQL/MX"?

**Comment by DaveBirdsall** [comment]
Looks good, Roberta. A step forward to an eventual successful Apache release.

**Comment by robertamarton** [comment]
Forgot to update .rat_excludes so the RAT sanity check failed.  Will do so shortly.

File: RAT_README.txt [hunk 2]
```diff
@@ -123,6 +123,13 @@ copyrights which are covered by MIT
  $MY_SQROOT/../sql/qmscommon/expat/* 
 
 
+The following files are open source from The Regents of the University of 
+California which are covered by BDS license
+
```
Comment (sandhyasun) [inline RAT_README.txt]: Typo. Should be BSD license ?

**Comment by sandhyasun** [comment]
Looks good for the most part if above issues are taken care of.

**Comment by arvind-narain** [comment]
The test failed because DCS_INSTALL_DIR didn't get set properly.

   ***INFO: Start of DCS install
   ***INFO: untarring build file /usr/lib/trafodion/traf_2015-11-24_0223/dcs-1.3.0.tar.gz
   ***INFO: modifying /home/jenkins/workspace/pyodbc_test-cdh/traf_run//conf/dcs-env.sh
   cat: dcs-env.sh: No such file or directory
   ***INFO: modifying /home/jenkins/workspace/pyodbc_test-cdh/traf_run//conf/dcs-site.xml
   cat: dcs-site.xml: No such file or directory
   ***INFO: creating /home/jenkins/workspace/pyodbc_test-cdh/traf_run//conf/servers file
   ***INFO: End of DCS install.

This pull request has removed the LICENSE file under dcs root directory the following in installer/dcs_installer is returning empty, hence DCS_DIR and subsequently DCS_INSTALL_DIR is not setup properly.

    # get the dcs install directory name which is imbedded in the tar file
    DCS_DIR=$(tar -tf $DCS_BUILD_FILE | grep LICENSE | sed -e "s@\(^.*\)/LICENSE.txt@\1@")
    DCS_DIR=$DCS_INSTALL_PATH/$DCS_DIR

Can we just use the tar filename to figure out the directory name ? Here are the current files in the root dcs directory:

  -bash-4.1$ cd dcs-1.3.0/
  -bash-4.1$ ls -l
  total 920
  drwxr-xr-x 3 trafodion trafodion   4096 Nov  2 18:51 bin
  drwxr-xr-x 2 trafodion trafodion   4096 Nov  2 18:51 conf
  -rw-r--r-- 1 trafodion trafodion 920659 Nov 24 02:15 dcs-1.3.0.jar
  drwxr-xr-x 5 trafodion trafodion   4096 Nov 24 02:14 dcs-webapps
  drwxr-xr-x 8 trafodion trafodion   4096 Nov 24 02:14 docs
  drwxrwxr-x 2 trafodion trafodion   4096 Nov 24 02:29 lib
  -bash-4.1$

**Comment by zellerh** [comment]
Looks good to me, thanks for doing all this work! Good to remove the TPC-D data generator help text. I looked around a bit and it seems the actual (copyrighted) TPC-D generator code was removed before open-sourcing the directory.

Question about DCS_INSTALL_DIR: Shouldn't we phase out this concept, given that DCS now always is located in $MY_SQROOT/../../dcs?

**Comment by hegdean** [comment]
No Hans,  We cannot phase it out as we use  DCS_INSTALL_DIR on our single node work spaces and it is not located in the same place as the cluster. We also use it in most of the start/stop/check scripts as well

**Comment by hegdean** [comment]
jenkins, retest

**Comment by zellerh** [comment]
Could we at least try $MY_SQROOT/../../dcs (or the place where it is located on the cluster) if DCS_INSTALL_DIR is not set?

**Comment by amandamoran** [comment]
I think my pull request needs to be merged and then this one will pass. 

Thanks!

**Comment by sureshsubbiah** [comment]
jenkins, retest

**Comment by robertamarton** [comment]
The last delivery to this pull request was to add Amanda's change from release 2.0 back to 1.3 so all the tests pass.  All the tests are now passing.  Anyone have objections on me committing this change?

**Comment by DaveBirdsall** [comment]
I say, Go for it!

**Pull request merged**